### PR TITLE
Fix #1588: Implement per-record encryption in configuration table

### LINKED ISSUE #1588: OIDC: Implement per-record encryption in configuration table
For activation using OIDC, we need to be able to encrypt the OIDC configuration using per-record encryption, due to storage of `client_secret` value required for accessing OAuth 2.0 tokens.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

**Comment by banterCZ** [comment]
- [ ] It has to be adapted to 1.9.0-SNAPSHOT changes

File: docs/db/changelog/changesets/powerauth-java-server/1.8.x/20240723-configuration-encryption.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.9.xsd">
+
+    <changeSet id="1" logicalFilePath="powerauth-java-server/1.8.x/20240723-configuration-encryption.xml" author="Lubos Racansky">
```
Comment (banterCZ) [inline docs/db/changelog/changesets/powerauth-java-server/1.8.x/20240723-configuration-encryption.xml:6]: @zcgandcomp Do we target OIDC to the backport of 1.8.x or 1.9.0?

Comment (banterCZ) [reply docs/db/changelog/changesets/powerauth-java-server/1.8.x/20240723-configuration-encryption.xml:6]: Ok, let's target it into 1.9.0

File: docs/PowerAuth-Server-1.9.0.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+For manual changes use SQL scripts:
+
+- [PostgreSQL script](./sql/postgresql/migration_1.8.0_1.9.0.sql)
+- [Oracle script](./sql/oracle/migration_1.8.0_1.9.0.sql)
+- [MSSQL script](./sql/mssql/migration_1.8.0_1.9.0.sql)
+
+
+### Add encryption_mode Column
+
+A new column `encryption_mode` has been added to the `pa_application_config` table to enable encryption of config values.
```
Comment (romanstrobl) [inline docs/PowerAuth-Server-1.9.0.md:19]: `enable encryption of config values` -> `enable encryption of configuration values`

**Review by romanstrobl — APPROVED** [review]
Great!

File: docs/PowerAuth-Server-1.9.0.md (hunk 2 above)
Comment (banterCZ) [reply docs/PowerAuth-Server-1.9.0.md:19]:
```suggestion
A new column `encryption_mode` has been added to the `pa_application_config` table to enable encryption of configuration values.
```

**Review by romanstrobl — DISMISSED** [review]
Looks OK.

**Review by romanstrobl — APPROVED** [review]

**Pull request merged**